mlir/lib/Dialect/Mesh/Transforms/Spmdization.cpp

```diff
@@ -450,53 +496,64 @@ reshardOn1DMesh(ImplicitLocOpBuilder &builder, MeshOp mesh,
       handlePartialAxesDuringResharding(builder, sourceSharding, targetSharding,
                                         sourceShard);
 
-  if (reducedSourceSharding == targetSharding) {
+  if (!force && reducedSourceSharding == targetSharding) {
```

Comment:
Why do we need `force`?

Reply:
This simply protects existing reshardings because they currently have no logic implemented to deal with `force`.

Reply:
We need the `force` for cases that require an `update_halo` (see below). Without `force` the sharding can potentially be elided.